paasta_tools/monitoring/check_capacity.py

```diff
@@ -0,0 +1,150 @@
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import paasta_print
+
+
+def parse_capacity_check_options():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--cpus', dest='cpus_default', type=float,
+        help='Default maximum cpu capacity before critical.',
+    )
+    parser.add_argument(
+        '--mem', dest='mem_default', type=float,
+        help='Default maximum memory capacity before critical.',
+    )
+    parser.add_argument(
+        '--disk', dest='disk_default', type=float,
+        help='Default maximum disk capacity before critical.',
+    )
+    parser.add_argument(
+        '--overrides', dest='overrides', type=str,
+        help='json file of per-attribute overrides.\n'
+        'In the format {attribute_name: {value: {cpus: num, disk: num, mem: num}}}',
+    )
+    parser.add_argument(
+        '--cluster', dest='cluster', type=str,
+        help='Cluster to check. Defaults to looking for the current cluster.',
+    )
+    parser.add_argument(
+        '--attributes', dest='attributes', type=str,
+        help='Comma separated list of attributes to check. If not specified, '
+        'these will be infered from the contents of the overrides file. '
+        'If overrides is also not specified, just check the whole cluster.\n'
+        'By default, only checks attributes individually; see --cross-product-check',
+    )
+    parser.add_argument(
+        '--cross-product-check', dest='cpc', action='store_true',
+        help='When specified, --cross-product-check will check combinations of '
+        'attributes instead of individual attributes. \n'
+        'eg if attributes is \'pool,region\', this will cause the check to fail '
+        'if any region-pool combination is below the threashold, or either\'s override',
+    )
+    options = parser.parse_args()
+
+    return options
+
+
+def calc_percent_usage(resource_item, value_to_check):
+    values = resource_item[value_to_check]
+    if values['total'] == 0:
+        return 0
+    return 100 * (values['used'] / values['total'])
+
+
+def run_capacity_check():
+    options = parse_capacity_check_options()
+    system_paasta_config = load_system_paasta_config()
+    cluster = options.cluster if options.cluster is not None else system_paasta_config.get_cluster()
+    to_check = filter(
+        lambda x: x[0] is not None,
+        [(options.disk_default, 'disk'), (options.cpus_default, 'cpus'), (options.mem_default, 'mem')],
+    )
+    if len(to_check) != 1:
+        paasta_print('UNKNOWN exactly one of --cpus, --mem, or --disk must be specified to capacity check')
+        sys.exit(3)
+    value_to_check = to_check[0][1]
+
+    client = get_paasta_api_client(cluster=options.cluster)
+    if client is None:
+        paasta_print('UNKNOWN Failed to load paasta api client')
+        sys.exit(3)
+
+    if options.overrides:
+        with open(options.overrides, 'r') as f:
```

Comment:
This is for PAASTA-10160.  So, for example, we could set a specific pool to only alert at 95% memory capacity, or a region to alert at 80%.